packages/next/server/web/spec-extension/cookies.ts

```diff
@@ -1,6 +1,11 @@
 import cookie from 'next/dist/compiled/cookie'
 import { CookieSerializeOptions } from '../types'
 
+type GetWithOptionsOutput = [
+  value: string | undefined,
+  options: { [key: string]: string } | undefined
+]
```

Comment:
It's just an opinionated API. 

The array way makes easy to get the cookies options separately:

```
const [value, options] = cookies.getWithOptions('foo')
```

rather than destructure from an object

```
const {value, ...options} = cookies.getWithOptions('foo')
```

Since there you are assuming the cookie value is under `value` key, that could not be evident.

Reply:
The configs of a cookie are enumerable, like `Path`, `Expires`, `HttpOnly`, etc. feel like the object return value is pretty convenient, since the `value` and `options` are not very different thing like a value and a setter

Reply:
You still have the same related issues. The cookies options are capitalized (like `Path`) so to be consistent the key for the value should be `Value`.

```js
const {Value, ...options} = cookies.getWithOptions('foo')
```

Still not seeing this is superior to just return an array

Reply:
If we do `{ value, options } = cookies.getWithOptions('foo')`, then there’s no problem related to capitalization right?

Reply:
oh ok, I miss that. Yep.